### PR TITLE
Add GitHub workflow and LuaCheck

### PR DESCRIPTION
**Pull request by Panquesito7**
Things added/changed:

- Add GitHub workflow and [LuaCheck](https://github.com/luarocks/luacheck).
- Fix warning reported by [LuaCheck](https://github.com/luarocks/luacheck).
- Add build status and license badge. [Build status](https://github.com/Panquesito7/meshport/actions) [License](https://www.gnu.org/licenses/lgpl-3.0.en.html)

### CODE REVIEW

File: .luacheckrc [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+read_globals = {
+    string = {fields = {"split", "trim"}},
+    table = {fields = {"copy", "getn", "indexof"}},
+
+    "minetest", "DIR_DELIM",
+    "vector", "VoxelArea",
+}
+
+ignore = {
+    "vertList",
```
Comment (random-geek) [inline .luacheckrc:14]: You can just remove `vertList` from the code, it's not used.

**Comment by random-geek** [comment]
Otherwise seems good. 👍

File: .luacheckrc (hunk 1 above)
Comment (Panquesito7) [reply .luacheckrc]: Done.

**Comment by random-geek** [comment]
Eh, that messed up more stuff. Please roll back the last commit, I can fix the other stuff later after this is merged.

**Comment by random-geek** [comment]
Thanks.

**Pull request merged**